ls1intum/Artemis
# `Exam mode`: Move combine template from generate student exam to prepare exercise start

### Checklist
#### General


#### Server


### Motivation and Context
This PR resolves some smaller issues in the server code of the exam mode.
I moved the operation combine template which is important for programming exercises to the action "prepare exercise start", because it does not really make sense to combine template commits when 

### Steps for Testing
- 1 Instructor
- 1 Exam with some registered users

1. Create the exam with a programming exercise and a quiz exercise
2. Commit some changes to the template repo 
3. Generate student exams
4. Make sure the history in the template was not yet changed
5. Click on prepare exercise start
6. Make sure the history in the template is combined into one commit, also for all participants
7. Reset the exam
8. Make sure the quiz exercise does not have a release date (potentially needs to be checked in the database)

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
no changes

### Screenshots
no changes
